mesa/riscv64.patch

```diff
@@ -1,44 +1,45 @@
 Index: PKGBUILD
 ===================================================================
---- PKGBUILD	(revision 444265)
+--- PKGBUILD	(revision 467369)
 +++ PKGBUILD	(working copy)
-@@ -6,3 +6,3 @@
+@@ -4,7 +4,7 @@
+ # Contributor: Andreas Radke <andyrtr@archlinux.org>
+ 
  pkgbase=mesa
--pkgname=('vulkan-mesa-layers' 'opencl-mesa' 'vulkan-intel' 'vulkan-radeon' 'vulkan-swrast' 'libva-mesa-driver' 'mesa-vdpau' 'mesa')
-+pkgname=('vulkan-mesa-layers' 'opencl-mesa' 'vulkan-radeon' 'libva-mesa-driver' 'mesa-vdpau' 'mesa')
+-pkgname=('vulkan-mesa-layers' 'opencl-mesa' 'vulkan-intel' 'vulkan-radeon' 'vulkan-swrast' 'vulkan-virtio' 'libva-mesa-driver' 'mesa-vdpau' 'mesa')
++pkgname=('vulkan-mesa-layers' 'opencl-mesa' 'vulkan-radeon' 'vulkan-swrast' 'vulkan-virtio' 'libva-mesa-driver' 'mesa-vdpau' 'mesa')
  pkgdesc="An open-source implementation of the OpenGL specification"
-@@ -14,3 +14,3 @@
-              'libomxil-bellagio' 'libclc' 'clang' 'libglvnd' 'libunwind' 'lm_sensors' 'libxrandr'
--             'systemd' 'valgrind' 'glslang' 'vulkan-icd-loader' 'directx-headers' 'cmake' 'meson')
-+             'systemd' 'valgrind' 'glslang' 'vulkan-icd-loader' 'cmake' 'meson')
- url="https://www.mesa3d.org/"
-@@ -41,4 +41,5 @@
-   # https://gitlab.freedesktop.org/mesa/mesa/-/issues/6229
--  CFLAGS+=' -mtls-dialect=gnu'
--  CXXFLAGS+=' -mtls-dialect=gnu'
-+  # riscv64 doesn't support -mtls-dialect=gnu
-+  # CFLAGS+=' -mtls-dialect=gnu'
-+  # CXXFLAGS+=' -mtls-dialect=gnu'
- 
-@@ -47,5 +48,5 @@
+ pkgver=22.3.3
+ pkgrel=3
+@@ -58,9 +58,9 @@
+     -D b_ndebug=true \
+     -D b_lto=false \
      -D platforms=x11,wayland \
--    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,iris,crocus,zink,d3d12 \
--    -D vulkan-drivers=amd,intel,swrast \
+-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,i915,iris,crocus,zink,d3d12 \
+-    -D vulkan-drivers=amd,intel,intel_hasvk,swrast,virtio-experimental \
 -    -D vulkan-layers=device-select,intel-nullhw,overlay \
-+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,swrast,zink \
-+    -D vulkan-drivers=amd \
++    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,zink,d3d12 \
++    -D vulkan-drivers=amd,swrast,virtio-experimental \
 +    -D vulkan-layers=device-select,overlay \
      -D dri3=enabled \
-@@ -53,3 +54,3 @@
+     -D egl=enabled \
      -D gallium-extra-hud=true \
--    -D gallium-nine=true \
-+    -D gallium-nine=false \
-     -D gallium-omx=bellagio \
-@@ -67,2 +68,3 @@
-     -D llvm=enabled \
-+    -D draw-use-llvm=false \
```

Comment:
draw-use-llvm was disabled because llvmpipe doesn't work properly. Have you tested that it works correctly now without the orcjit patch?

Reply:
Sorry for my carelessness. Since I don't have a headful device currently, I'll disable llvmpipe later.

Reply:
An easier test case could be just running `ssh -Y` or even xvfb for glxgears etc. Previously it just crashes with some LLVM errors.